app/src/test/java/com/codesoom/assignment/controllers/TaskControllerWebSpringDITest.java

```diff
@@ -0,0 +1,345 @@
+            void It_ThrowException() throws Exception {
+                mockMvc.perform(post(RESOURCE))
+                        .andExpect(status().isBadRequest());
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("PUT,PATCH /할 일/{id}는")
+    class Describe_PUT_PATCH{
+
+        @Nested
+        @DisplayName("{id}가 존재하지 않는다면")
+        class Context_NotExistsId{
+
+            @Test
+            @DisplayName("상태 405 , 해당 네트워크 메소드는 허용하지 않는 예외를 던진다")
+            void It_ThrowException() throws Exception {
+                mockMvc.perform(put(RESOURCE))
+                        .andExpect(status().isMethodNotAllowed());
+
+                mockMvc.perform(patch(RESOURCE))
+                        .andExpect(status().isMethodNotAllowed());
+            }
+        }
+
+        @Nested
+        @DisplayName("body가 존재하지 않는다면")
+        class Context_NotExistsBody{
+
+            private final String taskId = "1";
+
+            @Test
+            @DisplayName("상태 400 , 요청이 잘못 됐다는 예외를 던진다")
+            void It_ThrowException() throws Exception {
+                mockMvc.perform(put(RESOURCE +"/"+taskId))
+                        .andExpect(status().isBadRequest());
+
+                mockMvc.perform(patch(RESOURCE +"/"+taskId))
+                        .andExpect(status().isBadRequest());
+            }
+        }
+
+        @Nested
+        @DisplayName("{id}에 해당하는 할 일이 없다면")
+        class Context_InvalidId{
+
+            private final Task updateTask = new Task();
+            private final String invalidId = "1";
+
+            @BeforeEach
+            void setUp(){
+                service.clearTasks();
+                updateTask.setTitle("Update Title");
+            }
+
+            @Test
+            @DisplayName("상태 404 , 요청이 잘못 됐다는 예외를 던진다")
+            void It_ThrowException() throws Exception {
+                String content = mapper.writeValueAsString(updateTask);
+                mockMvc.perform(get(RESOURCE +"/"+ invalidId)
+                                .content(content))
+                        .andDo(print())
+                        .andExpect(status().isNotFound())
+                        .andExpect(content().string(containsString("Task not found")));
+            }
+        }
+
+        @Nested
+        @DisplayName("{id}와 body가 존재한다면")
+        class Context_ExistsIdAndBody{
+
+            private final Long updateId = 1L;
+            private final Task beforeTask = new Task();
+            private final Task afterTask = new Task(updateId , "after");
+
+            @BeforeEach
+            void setUp(){
+                service.clearTasks();
+                beforeTask.setTitle("before");
+                controller.create(beforeTask);
+            }
+
+            @Test
+            @DisplayName("body에 담긴 제목으로 수정한다")
+            void It_UpdateTask() throws Exception {
+                String content = mapper.writeValueAsString(afterTask);
+                mockMvc.perform(put(RESOURCE +"/"+updateId)
```

Comment:
```suggestion
                mockMvc.perform(put(RESOURCE + "/" + updateId)
```

코드를 일관성있게 정리하면 더 좋겠네요